bin.js

```diff
@@ -31,3 +34,12 @@ function getConfig() {
   }
 }
 
+function getRules(set) {
+  var configFileName = set ? ('eslint-config-' + set) : null
```

Comment:
Sorry, I'm not sure I'm following exactly what you are trying to do, but I think you are looking for this: http://eslint.org/docs/developer-guide/nodejs-api#getconfigforfile This will automatically calculate all of the config values for you (including `extends`), and it's available as public API. I would suggest not using internals directly as they do often change.